netplan_cli/cli/state_diff.py

```diff
@@ -0,0 +1,523 @@
+
+        if present_only_in_system:
+            iface[netdef_id]['netplan_state'].update({
+                'missing_search_domains': list(present_only_in_system),
+            })
+
+        if present_only_in_netplan:
+            iface[netdef_id]['system_state'].update({
+                'missing_search_domains': list(present_only_in_netplan),
+            })
+
+    def _analyze_mac_addresses(self, config: dict, iface: dict) -> None:
+        netdef_id = config.get('system_state', {}).get('id')
+        system_macaddress = config.get('system_state', {}).get('macaddress')
+        netplan_macaddress = config.get('netplan_state', {}).get('macaddress')
+
+        if system_macaddress and netplan_macaddress:
+            if system_macaddress != netplan_macaddress:
+                iface[netdef_id]['system_state'].update({
+                    'missing_macaddress': netplan_macaddress
+                })
+                iface[netdef_id]['netplan_state'].update({
+                    'missing_macaddress': system_macaddress
+                })
+
+    def _analyze_routes(self, config: dict, iface: dict) -> None:
+        netdef_id = config.get('system_state', {}).get('id')
+        netplan_routes = set(config.get('netplan_state', {}).get('routes', []))
+        system_routes = set(config.get('system_state', {}).get('routes', []))
+        netplan_routes = self._normalize_routes(netplan_routes)
+
+        # Filter out some routes that are expected to be added automatically
+        system_addresses = config.get('system_state', {}).get('addresses', [])
+        system_routes = self._filter_system_routes(system_routes, system_addresses)
+
+        present_only_in_netplan = netplan_routes.difference(system_routes)
+        present_only_in_system = system_routes.difference(netplan_routes)
+
+        if present_only_in_system:
+            iface[netdef_id]['netplan_state'].update({
+                'missing_routes': [route for route in present_only_in_system],
+            })
+
+        if present_only_in_netplan:
+            iface[netdef_id]['system_state'].update({
+                'missing_routes': [route for route in present_only_in_netplan],
+            })
+
+    def _analyze_missing_interfaces(self, report: dict) -> None:
+        netplan_interfaces = {iface for iface in self.netplan_state.netdefs}
+        system_interfaces_netdef_ids = {iface.netdef_id for iface in self.system_state.interface_list if iface.netdef_id}
+
+        netplan_only = netplan_interfaces.difference(system_interfaces_netdef_ids)
+        # Filtering out disconnected wifi netdefs
+        # If a wifi netdef is present in the netplan_only list it's because it's disconnected
+        netplan_only = list(filter(lambda i: self.netplan_state.netdefs.get(i).type != 'wifis', netplan_only))
+
+        system_only = []
+        for iface in self.system_state.interface_list:
+            # Let's not show the loopback interface as missing
```

Comment:
I didn't include it because in the most common cases, I believe, it would always show up as missing in Netplan. But now that NM can manage it (not sure if networkd can) maybe it'd make sense of include lo...